float32 tAx
float32 tAy
float32 tAz
float32 rAx
float32 rAy
float32 rAz
float32 tBx
float32 tBy
float32 tBz
float32 rBr
float32 rBy
float32 RBz
bool goal
---
bool goalReached
